Packs/GoogleApigee/Integrations/GoogleApigeeEventCollector/GoogleApigeeEventCollector.py

```diff
@@ -0,0 +1,310 @@
+        return token_response.get('access_token'), token_response.get('expires_in'), token_response.get('refresh_token')
+
+    @staticmethod
+    def is_token_valid(token_initiate_time: float, token_expiration_seconds: float) -> bool:
+        """
+        Check whether a token has expired. A token is considered expired if it reached its expiration date in
+        seconds minus a minute.
+
+        for example ---> time.time() = 300, token_initiate_time = 240, token_expiration_seconds = 120
+
+        300.0001 - 240 < 120 - 60
+
+        Args:
+            token_initiate_time (float): the time in which the token was initiated in seconds.
+            token_expiration_seconds (float): the time in which the token should be expired in seconds.
+
+        Returns:
+            bool: True if token has expired, False if not.
+        """
+        return time.time() - token_initiate_time < token_expiration_seconds - 60
+
+    def get_logs(self, from_date, to_time: int) -> Any:
+        res = self.http_request(
+            method='GET',
+            url_suffix=f'/v1/audits/organizations/{self.org_name}',
+            params={'startTime': from_date, 'endTime': to_time, 'expand': True}
+        )
+        return res
+
+
+def search_events(client, last_run: Dict[str, float], limit: int = 0) -> tuple[List[Dict[str, Any]], Dict[str, float]]:
+    """
+    Searches for logs using the '/<url_suffix>' API endpoint.
+    Args:
+        client (Client): client to interact with the service API
+        last_run (Dict): A list containing the time of the last run and the amount of events of this time
+        limit: int, the limit of the results to return. (is received only in zoom-get-events command)
+    Returns:
+        Tuple:
+            List: A list containing the events.
+            List: A dict containing the time of the last run and the amount of events of this time
+    """
+    to_time = int(time.time()) * MILLISECOENDS_CONVERT
+    last_fetch = last_run.get('last_fetch', to_time)
+    events_count = 0
+    logs_response = client.get_logs(last_fetch, to_time)
+    logs = logs_response.get('auditRecord', [])
+    if not logs:
+        return [], {'last_fetch': to_time, 'events_amount': events_count}
+    events_amount = last_run.get('events_amount', 0)
+    for event in reversed(logs):
+        if event.get('timeStamp') == last_fetch and events_amount > 0:
+            events_amount -= 1
+            logs.pop()
+        else:
+            break
+    if not limit:
+        limit = client.max_fetch
+    limit = 0 if len(logs) <= limit else -limit
+    events = logs[limit:]
+    time_stamp = events[0].get('timeStamp') if events else 0
+    if len(events) == client.max_fetch:
+        to_time = time_stamp
+    # could be less than limit and still same
+    if time_stamp == to_time:
+        events_count = sum(1 for event in events if event.get('timeStamp') == time_stamp)
```

Comment:
```suggestion
        events_count = sum(event.get('timeStamp') == time_stamp for event in events)
```
More pythonic (`True == 1`)